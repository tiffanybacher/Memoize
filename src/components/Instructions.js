import React, { Component } from 'react';

class Instructions extends Component {
  constructor(props) {
    super(props);
  }

  render() {
      let header;
      let instructions;

      if (this.props.answerIsShown && this.props.savedMsgIsShown) {
        header = 
          <h2>No problem! We will save this question to your Missed Flashcards!</h2>
        instructions = null;
      } else if (this.props.answerIsShown) {
        header =
          <h2>Check your answer!</h2>
        instructions =
          <p>If you feel good about your answer, let's keep going! If you think you may need more review, you can save this card to come back to this question later.</p>
      } else {
        header =
          <h2>Learn About Objects and Classes!</h2>
        instructions = 
          <p>Answer the question on the flashcard below. Submit your answer first then check your answer!</p>
      }

    return(
      <section className="Instructions">
        {header}
        {instructions}
      </section>
    );
  }
}

export default Instructions;