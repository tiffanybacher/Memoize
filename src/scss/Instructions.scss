.Instructions {

  h2 {
    margin: 0 0 20px 0;
  }

  .back-btn {
    text-decoration: underline;

    &:hover {
      color: $purple;
    }
  }
}
