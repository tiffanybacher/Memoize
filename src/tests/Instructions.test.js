import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Instructions from '../components/Instructions';

let mockAnswerIsShown = false;
let mockSavedMsgIsShown = false;

describe('Instructions', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Instructions
        answerIsShown={mockAnswerIsShown}
        savedMsgIsShown={mockSavedMsgIsShown} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});