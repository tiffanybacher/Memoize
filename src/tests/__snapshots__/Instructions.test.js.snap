// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions should match snapshot 1`] = `
<section
  className="Instructions"
>
  <h2>
    Learn About Objects and Classes!
  </h2>
  <p>
    Answer the question on the flashcard below. Submit your answer first then check your answer!
  </p>
</section>
`;
